docs/src/main/sphinx/develop/connectors.rst

```diff
@@ -102,6 +102,40 @@ an exception, use the ``TrinoException`` class with the ``INVALID_ORDER_BY``
 error code and an actionable message, to let users know how to write a valid
 query.
 
+WHERE clause pushdown
+^^^^^^^^^^^^^^^^^^^^^
+
+A ``WHERE`` clause can be processed faster if its constraints are pushed down
+to the underlying connectors.
+
+A constraint contains:
+* a summary of possible values (or ranges) for different columns,
+* an expression for pushing down function calls,
+* a map of assignments from variables in the expression to columns,
+* (optional) a predicate which tests a map of columns and their values,
+* (optional) a set of columns the predicate depends on, must be present if
+predicate is present.
+These should be treated as if joined with an AND operator.
+
+If a predicate is available, it is not weaker than the summary, and should
+have precedence over it. This overlap is due to historical reasons,
+as simple comparison pushdown was implemented first via summary, and more
+complex filters such as ``LIKE`` which required more expressive predicates
+were added later.
+
+The query optimizer may call ``applyFilter`` for a single query multiple times,
```

Comment:
Maybe it's worth adding a summary of the complete workflow (again, even if it'll be repeated in some other place), that is:
* the Trino planner calls methods of the `ConnectorMetadata` class, here that'll be `applyFilter`, that informs the engine what the connector can do, before any data is fetched
* `applyFitler` can process the arguments it receives, identifies which predicates it _can_ handle, and saves it in some form in the table handle object, which it returns
* the engine passes the table handle object to the page source provider or record set provider, which actually push down the predicates when fetching data from the data source

Reply:
I'll prepare an example and walk through it

Reply:
I'm not talking only about `applyFilter` examples but also describe what happens with the table handle later.